test-app/tests/integration/components/input-field-test.gts

```diff
@@ -0,0 +1,148 @@
+    await render(<template>
+      <InputField
+        @label="Label"
+        type="text"
+        @error="There is an error"
+        data-input
+        />
+    </template>);
+
+    const label = '[data-label]'
+    const input = '[data-input]';
+    const error = '[data-error]';
+   
+    const inputElement = document.querySelector('[data-input]');
+ 
+    const [id] = inputElement?.getAttribute('aria-describedby')?.trim().split(' ') ?? '';
+  
+    assert.dom(label).exists('Expected to have label component rendered');
+    assert.dom(label).hasText('Label', 'Expected to have label text "label"');
+
+    assert.dom(input).exists('Expected to have input tag rendered');
+    assert.dom(input).hasAttribute('type', 'text');
+   
+    assert.dom(error).exists('Expected to have error component rendered');
+    assert.dom(error).hasText('There is an error', 'Expected to have error text "error"');
+
+    if (id) {
+      assert.dom(error).hasAttribute('id', id);
+    }
+  });
+
+  test('it renders hint text', async function (assert) {
+    await render(<template>
+      <InputField
+        @label="Label"
+        type="text"
+        @hint="Hint text visible here"
+        data-input
+        />
+    </template>);
+
+    const label = '[data-label]'
+    const input = '[data-input]';
+    const hint = '[data-hint]';
+    
+    const inputElement = document.querySelector('[data-input]');
+ 
+    const [id] = inputElement?.getAttribute('aria-describedby')?.trim().split(' ') ?? '';
+
+    assert.dom(label).exists('Expected to have label component rendered');
+    assert.dom(label).hasText('Label', 'Expected to have label text "label"');
+
+    assert.dom(input).exists('Expected to have input tag rendered');
+    assert.dom(input).hasAttribute('type', 'text');
+   
+    assert.dom(hint).exists('Expected to have hint component rendered');
+    assert.dom(hint).hasText('Hint text visible here', 'Expected to have hint text "error"');
+
+    if (id) {
+     assert.dom(hint).hasAttribute('id', id);
+    }
+  });
+
+  test('it accepts @value and @onChange', async function (assert) {
+    assert.expect(8);
+    
+    const onChangeCallback = (value: string, e: Event | InputEvent) => {
+      assert.strictEqual(value, 'Banana', 'Expected input to match');
+      assert.ok(e, 'Expected `e` to be available as the second argument');
+      assert.ok(e.target, 'Expected direct access to target from `e`');
+      assert.step('handleChange');
+    };
+  
+    await render(<template>
+        <InputField
+          @label="Label"
+          type="text"
+          @value="Avocado"
+          @onChange={{onChangeCallback}}
+          data-input
+          />
+      </template>);
+
+    assert.verifySteps([]);
+
+    const input: HTMLInputElement | null  = document.querySelector('[data-input]');
+
+    if (!input) {
+      throw new Error('an input was not detected');
+    }
+    
+    assert.strictEqual(input.value, 'Avocado', 'input has the set @value');
```

Comment:
I would suggest to use `qunit-dom` here, which would also remove need to manually query the element. See https://github.com/CrowdStrike/ember-toucan-core/pull/36/files#diff-b7acb71cb9e9389fb6732d2b7eb21e52658721f8307ab8c972d27de292a9adb3R130